### PR TITLE
DeliveryCreatedEvent have to implement WebhookSerializableEventInterface

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
in order to comply with webhook functionality DeliveryCreatedEvent have to implement WebhookSerializableEventInterface

https://oat-sa.atlassian.net/browse/MS-22

- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

### CODE REVIEW

**Comment by boajer** [comment]
In addition to @julien-sebire comment - please take care about versioning too.

File: model/event/DeliveryCreatedEvent.php [hunk 1]
```diff
@@ -55,4 +55,22 @@ public function jsonSerialize()
             'delivery' => $this->deliveryUri
         ];
     }
+
+    /**
+     * @return string
+     */
+    public function getWebhookEventName()
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/event/DeliveryCreatedEvent.php]: if the WEBHOOK_EVENT_NAME is used only here, i think is safe to return the value,
if it is used inside others let's remove this function and change the visibility of the const to public
the extra code does not help to read faster the code

File: model/event/DeliveryCreatedEvent.php [hunk 2]
```diff
@@ -19,11 +19,11 @@
...
 namespace oat\taoDeliveryRdf\model\event;
 
-use JsonSerializable;
-use oat\oatbox\event\Event;
+use oat\tao\model\webhooks\WebhookSerializableEventInterface;
 
-class DeliveryCreatedEvent extends AbstractDeliveryEvent
+class DeliveryCreatedEvent extends AbstractDeliveryEvent implements WebhookSerializableEventInterface
 {
+    private const WEBHOOK_EVENT_NAME = 'DeliveryCreatedEvent';
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/event/DeliveryCreatedEvent.php]: please see my below comment about visibility

File: model/event/DeliveryCreatedEvent.php [hunk 3]
```diff
@@ -19,11 +19,11 @@
  */
 namespace oat\taoDeliveryRdf\model\event;
 
-use JsonSerializable;
-use oat\oatbox\event\Event;
+use oat\tao\model\webhooks\WebhookSerializableEventInterface;
 
-class DeliveryCreatedEvent extends AbstractDeliveryEvent
+class DeliveryCreatedEvent extends AbstractDeliveryEvent implements WebhookSerializableEventInterface
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/event/DeliveryCreatedEvent.php]: i think we are missing a test here

File: model/event/DeliveryCreatedEvent.php [hunk 4]
```diff
@@ -19,11 +19,11 @@
  */
 namespace oat\taoDeliveryRdf\model\event;
```
Comment (mike-ionut-mihai-sandu-tao) [inline model/event/DeliveryCreatedEvent.php]: version change required please don't forget about upadte and manifest,
also please apply the mandatory list provided by @Babacooll

**Comment by peetya** [comment]
Since you are relying on an interface from TAO, please be sure that the version range of the `tao` dependency is correct in the `manifest` file.

File: test/unit/model/event/DeliveryCreatedEventTest.php [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        $this->assertEquals($result, DeliveryCreatedEvent::class);
+    }
+
+    public function testGetWebhookEventName()
+    {
+        $event = new DeliveryCreatedEvent(self::DELIVERY_URI);
+        $result = $event->getWebhookEventName();
+        $this->assertEquals('DeliveryCreatedEvent', $result);
+    }
+}
```
Comment (boajer) [inline test/unit/model/event/DeliveryCreatedEventTest.php]: PSR-2 `All PHP files MUST end with a single blank line.` Could you fix please.

**Comment by bartlomiejmarszal** [comment]
I had to close it as branch was not correctly named

**Comment by mike-ionut-mihai-sandu-tao** [comment]
don't forget to add the new PR into this so the work can be tracked 
thanks

**Pull request closed without merging**